bundles/org.openhab.binding.pihole/src/main/java/org/openhab/binding/pihole/internal/PiHoleHandler.java

```diff
@@ -0,0 +1,264 @@
+        updateDecimalState(REPLY_OTHER_CHANNEL, localDnsStatistics.replyOther());
+        updateDecimalState(REPLY_DNSSEC_CHANNEL, localDnsStatistics.replyDNSSEC());
+        updateDecimalState(REPLY_NONE_CHANNEL, localDnsStatistics.replyNone());
+        updateDecimalState(REPLY_BLOB_CHANNEL, localDnsStatistics.replyBlob());
+        updateDecimalState(DNS_QUERIES_ALL_REPLIES_CHANNEL, localDnsStatistics.dnsQueriesAllTypes());
+        updateDecimalState(PRIVACY_LEVEL_CHANNEL, localDnsStatistics.privacyLevel());
+
+        var adsPercentageToday = localDnsStatistics.adsPercentageToday();
+        if (adsPercentageToday != null) {
+            var state = new QuantityType<>(new BigDecimal(adsPercentageToday.toString()), PERCENT);
+            updateState(ADS_PERCENTAGE_TODAY_CHANNEL, state);
+        }
+        updateState(ENABLED_CHANNEL, OnOffType.from(localDnsStatistics.enabled()));
+        if (localDnsStatistics.enabled()) {
+            updateState(DISABLE_ENABLE_CHANNEL, new StringType(ENABLE.toString()));
+        }
+    }
+
+    private void updateDecimalState(String channelID, @Nullable Integer value) {
+        if (value == null) {
+            return;
+        }
+        updateState(channelID, new DecimalType(value));
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(PiHoleActions.class);
+    }
+
+    @Override
+    public void dispose() {
+        adminService = null;
+        dnsStatistics = null;
+        var localScheduledFuture = scheduledFuture;
+        if (localScheduledFuture != null) {
+            try {
+                localScheduledFuture.cancel(true);
+            } catch (Exception e) {
+                logger.debug("Failed to cancel scheduled future", e);
+            } finally {
+                scheduledFuture = null;
+            }
+        }
+        super.dispose();
+    }
+
+    @Override
+    public Optional<DnsStatistics> summary() throws ExecutionException, InterruptedException, TimeoutException {
+        var local = adminService;
+        if (local == null) {
+            throw new IllegalStateException("AdminService not initialized");
+        }
+        return local.summary();
+    }
+
+    @Override
+    public void disableBlocking(long seconds) throws ExecutionException, InterruptedException, TimeoutException {
+        var local = adminService;
+        if (local == null) {
+            throw new IllegalStateException("AdminService not initialized");
+        }
+        local.disableBlocking(seconds);
+        // update the summary to get the value of DISABLED_CHANNEL channel
+        scheduler.schedule(this::update, HTTP_DELAY_SECONDS, SECONDS);
+        if (seconds > 0) {
+            // update the summary to get the value of ENABLED_CHANNEL channel
+            // after the X seconds it probably will be true again
+            scheduler.schedule(this::update, seconds + HTTP_DELAY_SECONDS, SECONDS);
```

Comment:
This smells a bit as it is depending on timing. Is there no other way?

Reply:
Why it smells bad? i.e. I'm disabling blocking for 5 min so after 5 min I would like to check the state (in theory it should get back to `blocking` state)

Reply:
A bit, not bad. ;-) I think i understand the case, not familiair with the API, so it might not be possible to improve.

If the blocking is cancelled/altered from pi-hole or something outside of openHAB it's state is not reflected in openHAB. It would be the best to have this event based, but that very much depends on the API possibilities. You might also poll every x seconds.
If all that is not possible, this would be ok to merge.